server-spi-private/src/main/java/org/keycloak/compatibility/CompatibilityMode.java

```diff
@@ -0,0 +1,19 @@
+package org.keycloak.compatibility;
+
+/**
+ * Exit code used by the CLI command.
+ */
+public enum CompatibilityMode {
+    ROLLING(CompatibilityResult.ROLLING_UPGRADE_EXIT_CODE),
```

Comment:
Can we reduce repetition by making the enum nested within the `CompatibilityResult` interface and using it as the canonical source of exit codes, i.e:

```java

interface CompatibilityResult {
   enum ExitCode {
      ROLLING(0),
      // see picocli.CommandLine.ExitCode
      // 1 -> software error
      // 2 -> usage error
      RECREATE(3),
      DISABLED(4);

      final int exitCode;

      ExitCode(int exitCode) {
         this.exitCode = exitCode;
      }

      public int getExitCode() {
         return exitCode;
      }
   }
}
```

Then we do `return CompatibilityResult.ExitCode.RECREATE.value();` when we need the int?

Reply:
`DISABLED(4)` does not make sense for the providers. Only 0 and 3 are valid exit codes, and the CLI command itself uses the other exit codes.